kernel/tests/write.rs

```diff
@@ -251,3 +270,356 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
     ));
     Ok(())
 }
+
+// set the value at .-separated `path` in `values` to `new_value` at `index`
+fn set_value(
+    mut value: &mut serde_json::Value,
+    path: &str,
+    new_value: serde_json::Value,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let parts: Vec<_> = path.split('.').collect();
+
+    for name in parts.iter().take(parts.len() - 1) {
```

Comment:
uhh, doesn't [`pointer_mut`](https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.pointer_mut) do exactly what you want here.

I'd suggest passing a "pointer" and then a key/value pair for what to set, so you don't have to mess with special casing the last bit of the path.